src/landing/LandingPage.js

```diff
@@ -1,14 +1,91 @@
 import React, { useState } from 'react';
+import { gql, useLazyQuery, useMutation } from '@apollo/client';
 import {
   Paper, Tab, Tabs,
 } from '@material-ui/core';
 import LoginForm from './LoginForm';
 import './styles.css';
 import RegisterForm from './RegisterForm';
 
+const GET_USERS = gql`
+  query Login{
+    users{
+      username
+      password
+    }
+  } 
+`;
+
+const CREATE_USER = gql`
+  mutation Register($username: String!, $firstName: String!, $lastName: String!, $password: String!){
+    createUser(username: $username, 
+      firstName: $firstName, 
+      lastName: $lastName, 
+      password: $password,
+      userRoleId: 3){
+         username,
+      }
+}
+`;
+
 const LandingPage = () => {
   const [form, setForm] = useState('Login');
 
+  const [getUsers, {
+    data: queryData, error: queryError,
+  }] = useLazyQuery(GET_USERS, {
+    notifyOnNetworkStatusChange: true,
+    fetchPolicy: 'cache-and-network',
+  });
+
+  const submitLogin = (username, password) => {
+    getUsers();
+
+    if (queryError) {
+      console.log(queryError);
+      alert('Something went wrong');
+      return;
+    }
+
+    let loginStatus = false;
+    if (queryData) {
+      queryData.users.forEach((user) => {
+        if (user.username === username
+            && user.password === password) {
+          loginStatus = true;
+        }
+      });
+    }
+
+    if (loginStatus === true) {
+      alert(`Login successful for ${username}`);
+    } else {
+      alert('Username or password incorrect!');
+    }
+  };
+
+  const [createUser, {
+    error: mutationError,
+  }] = useMutation(CREATE_USER);
+
+  const submitRegister = (firstName, lastName, username, password) => {
+    createUser({
+      variables: {
+        username, firstName, lastName, password,
+      },
+    })
+      .catch((err) => {
+        console.log(`Calling error: ${err}`);
+      });
+
+    if (mutationError) {
+      alert('Something went wrong...');
+      console.log(mutationError);
+    } else {
+      alert('Register successful! Redirecting to login...');
+    }
+  };
+
```

Comment:
Please check the useMutation hook API as we have better ways to handle errors or success (see onError, onComplete etc). https://www.apollographql.com/docs/react/data/mutations/

Reply:
updated